sdcm/tester.py

```diff
@@ -2750,7 +2750,12 @@ def get_email_data(self):  # pylint: disable=no-self-use
         return {}
 
     def all_nodes_scylla_shards(self):
-        all_nodes_shards = defaultdict(list)
+        all_nodes_shards = defaultdict(list, {'live_nodes': [], 'dead_nodes': []})
+
+        # There are scale tests with a lot of nodes. It will be so big list of nodes to print it in the email
+        if len(self.db_cluster.dead_nodes_list) + len(self.db_cluster.nodes) > 20:
```

Comment:
the check should be after you fill the lists...
and you should limit the list to 20